src/Halogen/HTML/Events.purs

```diff
@@ -100,6 +107,17 @@ onScroll = handler (EventType "scroll")
 onChange :: forall r i. (Event -> Maybe i) -> IProp (onChange :: Event | r) i
 onChange = handler ET.change
 
+onFileUpload :: forall r i t. Unfoldable t
+             => (t File -> Maybe i) -> IProp (onChange :: Event | r) i
+onFileUpload f = handler ET.change $
+  Event.target >=>
+  HTMLInputElement.fromEventTarget >=>
+  (HTMLInputElement.files >>> unsafePerformEffect) >=>
+  (items >>> pure) >=>
+  f
+  where unfoldEmpty :: forall f a. Unfoldable f => f a
```

Comment:
Looks like function `unfoldEmpty` has not been used.

Reply:
good catch